app/mutations/addresses/release.rb

```diff
@@ -31,18 +31,21 @@ def execute
       elsif address.address.to_host == @pool.gateway.to_host
         info "Skip gateway address=#{address.id} in pool=#{@pool.id}"
 
+      elsif ping?(address.address)
+        message = "Skip zombie address=#{address.id} in pool=#{@pool.id} that still responds to ping"
+        warn message
+        add_error(:address, :zombie, message)
       else
-        if ping?(address.address)
-          info "#{self.address} still responds to ping, deleting omitted"
-        else
-          info "Delete address=#{address.id} in pool=#{@pool.id}"
-          address.delete!
-        end
+        info "Delete address=#{address.id} in pool=#{@pool.id}"
+        address.delete!
       end
     end
 
+    # @param [IPAddr] ip address to ping
+    # @raise [RuntimeError] requires root privileges
+    #@return [Boolean]
     def ping?(ip_address)
-      icmp = Net::Ping::ICMP.new(ip_address.to_s)
+      icmp = Net::Ping::ICMP.new(ip_address.to_s, 0, 1)
```

Comment:
I suggest something more self-documenting like

```ruby
def ping?(ip_address, timeout: 1.0)
  port = 0
  icmp = Net::Ping::ICMP.new(ip_address.to_s, port, timeout)
  ...
```

This `0, 1` part is way too hard to figure out when reading.